packages/addons/_tests/_setup/suite.ts

```diff
@@ -44,8 +53,18 @@ export function setupTest<Addons extends AddonMap>(
 		});
 	}
 
-	vitest.beforeAll(({ name }) => {
-		const testName = path.dirname(name).split('/').at(-1)!;
+	const addonTestCases: Array<AddonTestCase<Addons>> = [];
+	for (const kind of options?.kinds ?? []) {
+		for (const variant of variants) {
+			const addonTestCase = { variant, kind };
+			if (!options?.filter || options?.filter?.(addonTestCase)) {
+				addonTestCases.push(addonTestCase);
+			}
+		}
+	}
+	let testName: string;
+	vitest.beforeAll(async ({ name }) => {
+		testName = path.dirname(name).split('/').at(-1)!;
 
 		// constructs a builder for create test projects
```

Comment:
```suggestion
		// constructs a builder to create test projects
```
unrelated, but while we are at it